Lib/typing.py

```diff
@@ -1526,40 +1463,6 @@ def __getitem__(self, params):
         return self.copy_with(params)
 
 
-class _UnionGenericAlias(_NotIterable, _GenericAlias, _root=True):
```

Comment:
Would there be code out in the wild that imports this class directly (despite it being private)? IIRC there are (or used to be?) use cases for importing `_GenericAlias`.

Reply:
How about we add back a dummy version, deprecate it, and work with some of those libraries to reduce their usage of it? I'm happy to help out with sending PRs to those projects. It should be pretty easy to avoid using it, I think, especially if we allow directly constructing `types.UnionType` (which this PR does)

Reply:
(FWIW, looks like typer is just vendoring pydantic)

Reply:
Yes, a plan like that sounds good.

Reply:
I pushed this, there's a basic version of `_UnionGenericAlias` again now.